### DISCUSSION TITLE
How to disable automatic scrolling of tab column when manually closing second-to-last tab?

### DISCUSSION THREAD

**Discussion by DavidAntliff** [piroor/treestyletab#3426] [Q&A]
It's of course possible to manually close a sequence of TST tabs by clicking the "X" slowly but repeatedly - the remaining tabs, below this position, will shift _up_ one tab position, placing the next tab's "X" under the cursor. A second click then closes what was the next tab. By repeating this, a long sequence of tabs can be closed, with the user reviewing each one as it goes by. This is useful when wanting to close a large number of tabs but with the opportunity to review each one before closing (just in case it's important).

There's an issue when the vertical column of tabs is longer than the browser screen height. On closing the second-to-last tab, the entire list of tabs will scroll itself _down_ such that the tab under the cursor when performing this sequence is *NOT* the next tab, but one some way above the original tab that was closed. Then the next click, _without moving the cursor_, closes the wrong tab. Of course if the user is paying attention, it can be restored, the mouse cursor moved, and the process begins again, but it is annoying.

Is there a way to disable the behaviour where closing the second-to-last tab suddenly causes the entire tree display to shift down?

EDIT: In my testing, it was always the second-to-last tab, but I've noticed since, in other windows, that sometimes it's the last tab, and sometimes it's a tab in the middle of the stack. It might be related to the current tab stack scroll position and the height of the tab currently being deleted - not really sure...

**Comment by piroor** [comment]
Is the box checked?: TST options => Development => All Configs => `simulateLockTabSizing`
When it is checked TST simulates "scroll position locking" behavior of Firefox's horizontal tab bar, and it will prevent accidental close of the previous tab on your case, like as this video clip:
https://github.com/piroor/treestyletab/assets/70062/a2b64f55-bfae-4f9e-a625-7c3ee8d31d2c
The checkbox is expected to be checked by default.

**Reply by DavidAntliff** [reply]
Thanks for the suggestion - in my case it **IS** ticked (i.e. checked).

**Reply by piroor** [reply]
Thank you for verifying. It looks a unknown issue so I've created a new issue #3524 to track this.
